internal/app/machined/pkg/adapters/secrets/encryption_salt.go

```diff
@@ -0,0 +1,39 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package secrets
+
+import (
+	"crypto/rand"
+	"io"
+
+	"github.com/siderolabs/talos/pkg/machinery/constants"
+	"github.com/siderolabs/talos/pkg/machinery/resources/secrets"
+)
+
+// EncryptionSalt adapter provides encryption salt generation.
+//
+//nolint:revive,golint
+func EncryptionSalt(r *secrets.EncryptionSaltSpec) encryptionSalt {
+	return encryptionSalt{
+		EncryptionSaltSpec: r,
+	}
+}
+
+type encryptionSalt struct {
+	*secrets.EncryptionSaltSpec
+}
+
+// Generate new encryption salt.
+func (a encryptionSalt) Generate() error {
+	buf := make([]byte, constants.DiskEncryptionSaltSize)
```

Comment:
i know systemd also extends the PCR with the disk uuid of the state, would it offer us anything extra?

Reply:
I'd also consider this. Basically some of those PCR extensions act as a "no way back" to make sure after you've used the TPM-backed secret you progress those PCRs further to make the secret inaccessible.

Reply:
I'm not sure how this is going to work for multiple volumes, and I think this is a bit different than this PR?